nncf/openvino/graph/model_transformer.py

```diff
@@ -526,3 +530,42 @@ def _apply_multiply_insertion_transformations(
                 destination_port.replace_source_output(multiply_node.output(0))
 
         return model
+
+    @staticmethod
+    def _apply_update_if_op_subgraph_transformations(
+        model: ov.Model, transformations: List[OVUpdateIfSubgraphCommand]
+    ) -> ov.Model:
+        """
+        Update model subgraph for IF node.
+
+        :param model: Model to update and insert a new subgraph.
+        :param transformations: Transformations with information of If node and an updated subgraph.
+        :return: Original model with an updated subgraph.
+        """
+        name_to_node_mapping = OVModelTransformer._get_name_to_node_mapping(model)
+        for transformation in transformations:
+            subgraph_model = transformation.subgraph_model
```

Comment:
Since `_apply_extract_if_subgraph_transformation` returns `return pointer to Model with sub-graph` I don't see any reason to change the sub-graph with the new one instead of updating it.

Reply:
This is the current limitation of the implementation of `OVModelTransformer`. `OVModelTransforemer` returns a new model for every applied transformation. The model after inserting FakeQuantizers is a copy, so there is a need to set the new model instead of an old one